In [1]:
import numpy as np
import pylab as pl

###NULL HYPOTHESIS: the % of former prisoners employed after release is the same or lower for candidates who participated in the program as for the control group, significance level p=0.05

###what if we used the values for where the former inmate was or was not ever employed in a CEO transitional job 3 years after release (4th row of the table)?

* P0 = 0.521
* P1 = 0.533

In [2]:
alpha=0.05
#we like fractions better then percentages. as a rule of thumb, either use fractions or counts
P_0=52.2*0.01 
P_1=53.3*0.01

n_0=409
n_1=564

#lets get the counts by multiplying by the sample size
Nt_0=P_0*n_0
Nt_1=P_1*n_1

## z test

In [3]:
# Z-test

#define the sample proportion first
sp=(P_0*n_0+P_1*n_1)/(n_1+n_0)
print sp

0.528376156218


In [6]:
# standard deviation function
sp_stdev= lambda p, n: np.sqrt( p * ( 1 - p ) /n[0] +  p * ( 1 - p )/n[1]  )

sp_stdev_3y=sp_stdev((Nt_0+Nt_1)/(n_0+n_1),[n_0,n_1])
print P_0, n_0, n_1, sp_stdev_3y

0.522 409 564 0.0324208723673


In [7]:
# z score calculation
zscore = lambda p0, p1, s : (p0-p1)/s
z_3y = zscore(P_1, P_0, sp_stdev_3y)
print z_3y

0.339287600759


In [9]:
##p-value for employment not ever employed in a CEO transitional job 3 years after: 
#since the largest number we read off the table for alpha=0.05 is smaller then the value for our statistic our p-value will be smaller then calculated using .9998
##thus using 0.6331 

p_3y=1-0.6337


def report_result(p,a):
    print 'is the p value {0:.2f} smaller than the critical value {1:.2f}? '.format(p,a)
    if p<a:
        print "YES!"
    else: print "NO!"
    
    print 'the Null hypothesis is {}'.format( 'rejected' if p<a  else 'not rejected') 

    
report_result(p_3y,alpha)

is the p value 0.37 smaller than the critical value 0.05? 
NO!
the Null hypothesis is not rejected


##Chi-square test

###About the employment for >6 consecutive months

#####List of Data
* test sample_employed = 14.7 * 5.64
* test sample_unemployed = 85.3 * 5.64
* control sample_employed = 11.9 * 4.09
* control sample_unemploye = 88.1 * 4.09
* total_employed = 131.579
* total_unemployed = 841.421

In [11]:
Ntot = 973
expected = 564*409*131.579*841.421
sample_values = [[14.7*5.64,85.3*5.64],[11.9*4.09,88.1*4.09]]
 
chisqstat= lambda N, values, expect : N*((values[0][0]*values[1][1]-values[0][1]*values[1][0])**2)/(expect)

print chisqstat(Ntot,  sample_values, expected)

1.58939333193


* 1.58939333193 is not larger than 3.84 (using percentage points of the Chi-square distribution)
* 3.84 from 0.05 probability and 1 degrees of freedom
* From 1.58939333193 < 3.84, the Null hypothesis is not rejected.